File: Sunrin-ICPC-2022/Final-H/FH.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

function gcd(a, b){ return b ? gcd(b, a % b) : a; }

const tc = +input[0];
for(let i=1; i<=tc; i++){
    const n = +input[i];
    if(n % 2 == 1) console.log(n * (n-1) * (n-2));
    else if(gcd(n, n-3) ==  1) console.log(n * (n-1) * (n-3));
    else console.log((n-1) * (n-2) * (n-3));
}

File: Sunrin-ICPC-2022/Final-H/FH.py
def gcd(a, b):
    while b != 0: (a,b) = (b,a%b)
    return a

tc = int(input())

for t in range(tc):
    n = int(input())
    if n % 2: print(n * (n-1) * (n-2))
    elif gcd(n, n-3) == 1: print(n * (n-1) * (n-3))
    else: print((n-1) * (n-2) * (n-3))